### PR TITLE
Progress docs

### PR DESCRIPTION
**Pull request by lilyvc**
Add initial docs page for Progress.

### CODE REVIEW

File: packages/lab/stories/progress/progress.doc.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+## Examples
+
+### CircularProgress
+
+<Canvas>
+  <Story id="lab-CircularProgress--default" />
+</Canvas>
+
+### LinearProgress
```
Comment (libertymayc) [inline packages/lab/stories/progress/progress.doc.stories.mdx]:
```suggestion
### Linear Progress
```

File: packages/lab/stories/progress/progress.doc.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+# Progress
+
+A progress indicator gives the user an understanding of how long a system operation will take. It should be used when the operation will take more than a second to complete. Two variants are available to accommodate different layout constraints: Linear and Circular.
+
+## Examples
+
+### CircularProgress
+
+<Canvas>
+  <Story id="lab-CircularProgress--default" />
```
Comment (libertymayc) [inline packages/lab/stories/progress/progress.doc.stories.mdx:25]:
```suggestion
  <Story id="lab-progress-circular--default" />
```
this is the url / path

File: packages/lab/stories/progress/progress.doc.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+</Canvas>
+
+### LinearProgress
+
+<Canvas>
+  <Story id="lab-LinearProgress--default" />
```
Comment (libertymayc) [inline packages/lab/stories/progress/progress.doc.stories.mdx:31]:
```suggestion
  <Story id="lab-progress-linear--default" />
```

File: packages/lab/stories/progress/progress.doc.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+</Canvas>
+
+### With maximum value
+
+Specify the maximum value of the progress indicator. The percentage progress will be calculated using this value. Default value is 100.
+
+#### CircularProgress with maximum value
+
+<Canvas>
+  <Story id="lab-CircularProgress--max-value" />
```
Comment (libertymayc) [inline packages/lab/stories/progress/progress.doc.stories.mdx:41]:
```suggestion
  <Story id="lab-progress-circular--max-value" />
```

File: packages/lab/stories/progress/progress.doc.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+<Canvas>
+  <Story id="lab-LinearProgress--max-value" />
+</Canvas>
+
+### With progressing value
+
+#### CircularProgress with progressing value
+
+<Canvas>
+  <Story id="lab-CircularProgress--progressing-value" />
```
Comment (libertymayc) [inline packages/lab/stories/progress/progress.doc.stories.mdx:55]:
```suggestion
  <Story id="lab-progress-linear--progressing-value" />
```

File: packages/lab/stories/progress/progress.doc.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+</Canvas>
+
+#### LinearProgress with progressing value
+
+<Canvas>
+  <Story id="lab-LinearProgress--progressing-value" />
```
Comment (libertymayc) [inline packages/lab/stories/progress/progress.doc.stories.mdx:61]:
```suggestion
  <Story id="lab-progress-linear--progressing-value" />
```

File: packages/lab/stories/progress/progress.doc.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+import { CircularProgress } from "@salt-ds/lab";
+```
+
+```
+import { LinearProgress } from "@salt-ds/lab";
+```
+
+### Props
+
+<ArgsTable of={CircularProgress} />
```
Comment (libertymayc) [inline packages/lab/stories/progress/progress.doc.stories.mdx:78]: add linear too

Comment (libertymayc) [reply packages/lab/stories/progress/progress.doc.stories.mdx:78]: and characteristics etc

File: packages/lab/stories/progress/progress.doc.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,112 @@
+import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
+import { CircularProgress, LinearProgress } from "@salt-ds/lab";
+import {
+  CharacteristicUsage,
+  CSSClassTable,
+} from "css-variable-docgen-components";
+
+<Meta
+  title="Documentation/Lab/Progress"
+  component={CircularProgress}
```
Comment (libertymayc) [inline packages/lab/stories/progress/progress.doc.stories.mdx:10]: I wonder if we should split this out into Documentation/Lab/Progress/Linear and Documentation/Lab/Progress/Circular to follow how the stories are, and also fix this being one component of two

Comment (lilyvc) [reply packages/lab/stories/progress/progress.doc.stories.mdx:10]: yeah I thought about this, not sure about how to structure it but I can take a look

File: packages/lab/stories/progress/progress.linear.doc.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,70 @@
+import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
+import { LinearProgress, LinearProgress } from "@salt-ds/lab";
```
Comment (libertymayc) [inline packages/lab/stories/progress/progress.linear.doc.stories.mdx:2]:
```suggestion
import { LinearProgress } from "@salt-ds/lab";
```

File: packages/lab/stories/progress/progress.circular.doc.stories.mdx [hunk 10]
```diff
@@ -0,0 +1,70 @@
+import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
+import { CircularProgress, LinearProgress } from "@salt-ds/lab";
```
Comment (libertymayc) [inline packages/lab/stories/progress/progress.circular.doc.stories.mdx:2]:
```suggestion
import { CircularProgress } from "@salt-ds/lab";
```

File: packages/lab/stories/progress/progress.linear.doc.stories.mdx (hunk 9 above)
Comment (lilyvc) [reply packages/lab/stories/progress/progress.linear.doc.stories.mdx]: thanks :)

**Review by libertymayc — APPROVED** [review]

**Review by libertymayc — APPROVED** [review]

**Comment by alycrys** [comment]
I feel that we could have one file for both types of progress, the docs are literally the same

**Comment by lukeac123** [comment]
yeh i agree i think having as one doc but displaying the two different examples under each section makes sense. I'm happy with this as it is in labs but maybe consolidate to one when promoting to core

**Pull request merged**